### ISSUE TITLE
Possible bug in returning a switch expression

### ISSUE THREAD

**Issue by gwhitney** [DanielXMoore/Civet#701]
It appears that you can mix explicit returns with return value assignment:
```
function foo(a)
   if bar[a] === 'three' then return 3
   return .= 9
   return.value *= a
```
compiles to
```
function foo(a) {
  if (bar[a] === "three") return 3;
  let ret = 9;
  ret *= a;
  return ret;
}
```
which works as expected. And it appears you can return a switch expression:
```
function foo(a)
   if bar[a] === 'three' then return 3
   if a < 0 then return switch bar[-a]
      'three' 3
   9
```
compiles to
```
function foo(a) {
  if (bar[a] === "three") return 3;
  if (a < 0)
    return (() => {
      let m;
      if (((m = bar[-a]), m === "three")) {
        return 3;
      }
    })();
  return 9;
}
```
which is also OK. But if you mix all these things, e.g. in
```
function foo(a)
   if bar[a] === 'three' then return 3
   if a < 0 then return switch bar[-a]
      'three' 3
   return .= 9
   return.value *= a
```
you then get from compilation the TypeScript code
```
function foo(a) {
  if (bar[a] === "three") return 3;
  if (a < 0)
    return (() => {
      let m;
      if (((m = bar[-a]), m === "three")) {
        return ret;
      }
    })();
  let ret = 9;
  ret *= a;
  return ret;
}
```
which fails to compile under tsc with `TS2454: Variable 'ret' is used before being assigned.` and fails to run in the Playground with `ReferenceError: can't access lexical declaration 'ret' before initialization`. Seems like a bug -- by analogy with the case without the return value assignment, it seems like the first occurrence of `return ret;` in this last compiled version should be `return 3;`. Unless I am trying to do something I shouldn't be able to?

**Comment by edemaine** [comment]
That definitely looks like a bug. Thanks for the report! Seems like the `return` value magic is polluting the IIFE of the `switch` somehow.

**Comment by STRd6** [comment]
Fixed in v0.6.32
